TESTS/mbed_platform/FileHandle/main.cpp

```diff
@@ -0,0 +1,532 @@
+ *       fputc return written element
+ *       on failure, returns EOF and stream error is sets
+ *
+ *  When read previously written data from file
+ *  Then underneath retargeting layer read function is called
+ *       fgetc return read element
+ *       read data match previously written
+ *       on failure, returns EOF and stream error/eof is sets respectively
+ *
+ */
+void test_fputc_fgetc()
+{
+    std::FILE *file;
+    const uint32_t FS = 3;
+    TestFile<FS> fh;
+    char char_buf[3] = {'a', 'b', 'c' };
+    int ret;
+
+    file = fdopen(&fh, "w+");
+    TEST_ASSERT_NOT_NULL(file);
+#ifdef __ICCARM__
+    std::setbuf(file, NULL); // fix IAR every byte sync
+#else
+    mbed_set_unbuffered_stream(file);
+#endif
+
+    // write 1; expected written 1
+    TestFile<FS>::resetFunctionCallHistory();
+    ret = std::fputc(char_buf[0], file);
+    TEST_ASSERT_TRUE(TestFile<FS>::functionCalled(TestFile<FS>::fnWrite));
+    TEST_ASSERT_EQUAL_INT(char_buf[0], ret);
+
+    // write 1; expected written 1
+    TestFile<FS>::resetFunctionCallHistory();
+    ret = std::fputc(char_buf[1], file);
+    TEST_ASSERT_TRUE(TestFile<FS>::functionCalled(TestFile<FS>::fnWrite));
+    TEST_ASSERT_EQUAL_INT(char_buf[1], ret);
+
+    // write 1; expected written 1
+    TestFile<FS>::resetFunctionCallHistory();
+    ret = std::fputc(char_buf[2], file);
+    TEST_ASSERT_TRUE(TestFile<FS>::functionCalled(TestFile<FS>::fnWrite));
+    TEST_ASSERT_EQUAL_INT(char_buf[2], ret);
+
+#ifndef __ICCARM__ // prevents IAR infinite loop
+    // write 1; expected written 0
+    TestFile<FS>::resetFunctionCallHistory();
+    ret = std::fputc(char_buf[0], file);
+    TEST_ASSERT_TRUE(TestFile<FS>::functionCalled(TestFile<FS>::fnWrite));
+    TEST_ASSERT_TRUE(std::ferror(file) > 0);
+    std::clearerr(file); // for ARMCC
+    TEST_ASSERT_EQUAL_INT(EOF, ret);
+#endif
+
+    fseek(file, 0, SEEK_SET);
+
+    // read 1; expected read 1
+    TestFile<FS>::resetFunctionCallHistory();
+    ret = std::fgetc(file);
+    TEST_ASSERT_TRUE(TestFile<FS>::functionCalled(TestFile<FS>::fnRead));
+    TEST_ASSERT_EQUAL_INT(char_buf[0], ret);
+
+    // read 1; expected read 1
+    TestFile<FS>::resetFunctionCallHistory();
+    ret = std::fgetc(file);
+#ifndef __ICCARM__
+    // IAR optimize reads
+    TEST_ASSERT_TRUE(TestFile<FS>::functionCalled(TestFile<FS>::fnRead));
+#endif
+    TEST_ASSERT_EQUAL_INT(char_buf[1], ret);
+
+    // read 1; expected read 1
+    TestFile<FS>::resetFunctionCallHistory();
+    ret = std::fgetc(file);
+#ifndef __ICCARM__
+    // IAR optimize reads
+    TEST_ASSERT_TRUE(TestFile<FS>::functionCalled(TestFile<FS>::fnRead));
+#endif
+    TEST_ASSERT_EQUAL_INT(char_buf[2], ret);
+
+#ifndef __ICCARM__
+    // read 1; expected read 0
+    TestFile<FS>::resetFunctionCallHistory();
+    ret = std::fgetc(file);
+    TEST_ASSERT_TRUE(TestFile<FS>::functionCalled(TestFile<FS>::fnRead));
+    TEST_ASSERT_TRUE(std::feof(file) > 0);
```

Comment:
Again, non-zero. Not specified as positive.

Reply:
fixed